view/frontend/web/js/product/options.js

```diff
@@ -59,9 +59,26 @@ define([
                             break;
                         case 'select':
                             if (val && $(this).hasClass('datetime-picker')) {
+
                                 var role = $(this).data('calendar-role');
                                 skuSuffix += '***' + identifier + '===' + role + '===' + val;
-                                price += document.round(configOptions.price)
+
+                                if(!skipIds.includes(optionId)){
+
+                                    var dateSelected = true;
+                                    $('[id^=options_' + optionId + '_]').each(function () {
+                                        if($(this).val() === ''){
+                                            dateSelected = false;
+                                        }
+                                    });
+
+                                    if(dateSelected){
+                                        price += document.round(configOptions.price);
+                                    }
```

Comment:
is this fixing an issue that had been present in the code already that you just found now? Or was this case covered before, but with the validation changes, you needed to add this code in to maintain the same level of functionality?

Reply:
This case wasn't covered before, because date select consists of 3+ fields, check is needed not to have price increase until all fields are selected.